lib/ConstExtract/ConstExtract.cpp

```diff
@@ -483,34 +487,40 @@ ConstValueTypeInfo ConstantValueInfoRequest::evaluate(
 
   std::vector<ConstValueTypePropertyInfo> Properties;
   llvm::Optional<std::vector<EnumElementDeclValue>> EnumCases;
-
-  if (shouldExtract(Decl)) {
-    // Use 'getStoredProperties' to get lowered lazy and wrapped properties
-    auto StoredProperties = Decl->getStoredProperties();
-    std::unordered_set<VarDecl *> StoredPropertiesSet(StoredProperties.begin(),
-                                                      StoredProperties.end());
-    for (auto Property : StoredProperties) {
+  std::function<void(class Decl *)> extract;
+
+  // Use 'getStoredProperties' to get lowered lazy and wrapped properties.
+  // @_objcImplementation extensions might contain stored properties.
+  auto StoredProperties = Decl->getStoredProperties();
+  std::unordered_set<VarDecl *> StoredPropertiesSet(StoredProperties.begin(),
+                                                    StoredProperties.end());
+  for (auto Property : StoredProperties) {
+    if (shouldExtract(Property->getDeclContext())) {
       Properties.push_back(extractTypePropertyInfo(Property));
     }
+  }
+
+  extract = [&](class Decl *Member) {
+    // Ignore plain stored properties collected above,
+    // instead gather up remaining static and computed properties.
+    if (auto *VD = dyn_cast<VarDecl>(Member))
+      if (!StoredPropertiesSet.count(VD))
+        Properties.push_back(extractTypePropertyInfo(VD));
+    // Extract from macro expanded members
+    Member->visitAuxiliaryDecls(extract);
+  };
 
+  if (shouldExtract(Decl)) {
     for (auto Member : Decl->getMembers()) {
```

Comment:
It's possible to use `getAllMembers()` here instead of `visitAuxiliaryDecls`, but then instead of checking `shouldExtract` for the declaration just once, will need to check `shouldExtract` for every single member 🤔

Reply:
I think `visitAuxiliaryDecls` is just fine. 👍🏼

Reply:
Now I've thought about it more, actually calling getAllMembers behaves the same since ConstExtract runs at the very end of pipeline. Switching to use getAllMembers to simplify code a bit